source/HailPanel.h

```diff
@@ -58,9 +58,11 @@ class HailPanel : public Panel {
 	PlayerInfo &player;
 	std::shared_ptr<Ship> ship = nullptr;
 	std::function<void(const Government *)> bribeCallback = nullptr;
+	const StellarObject *object = nullptr;
 	const Planet *planet = nullptr;
 	const Sprite *sprite = nullptr;
```

Comment:
we dont need this anymore, do we?

Reply:
Removing them would require a more significant refactor than I think is necessary to fix the bug. I might make some larger changes later. Leaving them here doesn't really cost anything (performance isn't a particularly high priority in the hail panel) and provides more convenient access on member methods.

Reply:
sprite is only used in one other occasion, I am only referring to sprite, not to planet

Reply:
Looks like you're right. I've removed HailPanel::sprite.